src/main/java/org/zalando/nakadi/service/CursorsService.java

```diff
@@ -80,11 +81,37 @@ public CursorsService(final ZooKeeperHolder zkHolder,
 
         validateStreamId(cursors, streamId, subscriptionId);
 
+        LOG.debug("[COMMIT_CURSORS] stream IDs validation finished");
+
+        final Iterator<EventTypePartition> distinctPartitions = cursors.stream()
```

Comment:
In this way you can get the map, etname_partition -> cursor with max offset per partition:

`        cursors.stream()
                .collect(Collectors.groupingBy((s -> s.getEventType() + "_" + s.getPartition()))).entrySet().stream()
                .map(entry -> cursors.stream().max(Comparator.comparingLong(sc -> Long.valueOf(sc.getOffset()))));`

After that you can just commit these cursors.